src/meta/meta_track.c

```diff
@@ -265,10 +265,12 @@ __wt_meta_track_off(WT_SESSION_IMPL *session, bool need_sync, bool unroll)
     }
 
     /*
-     * If we don't have the metadata cursor (e.g, we're in the process of creating the metadata), we
-     * can't sync it.
+     * If we don't have the metadata cursor or there is no meta checkpoint in the session (e.g,
+     * we're in the process of creating the metadata), we can't sync it.
      */
-    if (!need_sync || session->meta_cursor == NULL || F_ISSET(S2C(session), WT_CONN_IN_MEMORY))
+    if (!need_sync || session->meta_cursor == NULL || F_ISSET(S2C(session), WT_CONN_IN_MEMORY) ||
```

Comment:
I am personally scared of this change. I didn't give much thought around the possible behaviours around this. Please make sure if this is fine.

Reply:
So this change was necessary because we are performing drop from `__metadata_load_hot_backup` and I believe we haven't initialised `(S2C(session)->meta_ckpt_session` yet. Therefore I added the the extra check if `(S2C(session)->meta_ckpt_session == NULL && !FLD_ISSET(S2C(session)->log_flags, WT_CONN_LOG_ENABLED))` to avoid this use case.